### DISCUSSION TITLE
Why my TurboAWGNsystem's output is all zero

### DISCUSSION THREAD

**Discussion by Sean-xiang** [NVlabs/sionna#405] [Q&A]
I'm using Turbo encoder and decoder to simulate text transmission , but the trubo decoder's output is all zero .
Please tell me how to fix it ?
![code1](https://github.com/NVlabs/sionna/assets/55120589/18319fcb-f172-408e-abf3-e16067ea08e1)
![code2](https://github.com/NVlabs/sionna/assets/55120589/23b1c5cc-b777-43b8-8810-618eb80ed5b2)
![result1](https://github.com/NVlabs/sionna/assets/55120589/b982d2eb-0746-451a-9ca5-9591a85cb447)

**Comment by merlinND** [comment]
Hello @Sean-xiang,

Could you please share the full source for your decoder and a minimal script that you use to test it?
These partial screenshots do not provide enough information.

**Comment by SebastianCa** [comment]
Hi @Sean-xiang,

As @merlinND mentioned, it’s difficult to identify the issue without seeing the entire code. However, it might be helpful to check the SNR values used in your simulation. There could be a clipping issue with the LLRs. Can you please check the values of `llr`?
